examples/with-auth/routes/auth.js

```diff
@@ -0,0 +1,202 @@
+    // always refetch session info after N seconds has elapsed since last
+    // checked. Sensible values are between 0 (always check the server) and a
+    // few minutes.
+    clientMaxAge = 60000,
+    // URL of the server (e.g. 'http://www.example.com'). Used when sending
+    // sign in links in emails. Autodetects to hostname if null.
+    serverUrl = null,
+    // Mailserver configuration for nodemailer (defaults to localhost if null)
+    mailserver = null
+  } = {}) => {
+  if (app === null) {
+    throw new Error('app option must be a next server instance')
+  }
+
+  if (server === null) {
+    throw new Error('server option must be an express server instance')
+  }
+
+  if (User === null) {
+    throw new Error('user option must be a User model')
+  }
+
+  // Load body parser to handle POST requests
+  server.use(bodyParser.json())
+  server.use(bodyParser.urlencoded({extended: true}))
+
+  // Configure sessions
+  server.use(session({
+    secret: secret,
+    store: store,
+    resave: false,
+    rolling: true,
+    saveUninitialized: true,
+    httpOnly: true,
+    cookie: {
+      maxAge: maxAge
+    }
+  }))
+
+  // Add CSRF to all POST requests
+  // (If you want to add exceptions to paths you can do that here)
+  server.use((req, res, next) => {
+    csrf(req, res, next)
+  })
+
+  // With sessions connfigured (& before routes) we need to configure Passport
+  // and trigger passport.initialize() before we add any routes
+  passportStrategies.configure({
+    app: app,
+    server: server,
+    user: User
+  })
+
+  // Add route to get CSRF token via AJAX
+  server.get(path + '/csrf', (req, res) => {
+    return res.json({csrfToken: res.locals._csrf})
+  })
+
+  // Return session info
+  server.get(path + '/session', (req, res) => {
+    let session = {
+      clientMaxAge: clientMaxAge,
+      csrfToken: res.locals._csrf
+    }
+
+    // Add user object to session if logged in
+    if (req.user) {
+      session.user = req.user
+    }
+
+    return res.json(session)
+  })
+
+  // On post request, redirect to page with instrutions to check email for link
+  server.post(path + '/email/signin', (req, res) => {
+    const email = req.body.email || null
+
+    if (!email || email.trim() === '') {
+      return app.render(req, res, pages + '/signin', req.params)
+    }
+
+    const token = uuid()
+    const verificationUrl = (serverUrl || 'http://' + req.headers.host) + path + '/email/signin/' + token
+
+    // Create verification token save it to database
+    // @FIXME Improve error handling
+    User.one({email: email}, function (err, user) {
+      if (err) {
+        throw err
+      }
+      if (user) {
+        user.token = token
+        user.save(function (err) {
+          if (err) {
+            throw err
+          }
+        })
+      } else {
+        User.create({email: email, token: token}, function (err) {
+          if (err) {
+            throw err
+          }
+
+          nodemailer
```

Comment:
Shouldn't this execute whether or not the user exists? As it is now,  a new token is generated and saved for the existing user, but he is not getting the updated verification url as a token, so the old link he got will no longer work.